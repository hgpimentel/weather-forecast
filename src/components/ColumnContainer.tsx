import styled from "styled-components";

const ColumnContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

export default ColumnContainer;
