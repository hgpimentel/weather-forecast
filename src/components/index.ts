export { default as Layout } from "./Layout/Layout";
export { default as ColumnContainer } from "./Container/ColumnContainer";
export { default as CenteredContainer } from "./Container/CenteredContainer";
export { default as NotFound } from "./NotFound/NotFound";
