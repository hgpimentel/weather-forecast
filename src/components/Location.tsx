import React, { useContext } from "react";
import ColumnContainer from "./ColumnContainer";
import CenteredContainer from "./CenteredContainer";
import LocationContext from "../store/location-context";
import styled from "styled-components";

const CustomColumnContainer = styled(ColumnContainer)`
  padding: 2rem 3rem;

  & > div:last-child {
    margin-top: 1rem;
  }
`;

const Location: React.FC = () => {
  const { currentLocation } = useContext(LocationContext);

  if (!currentLocation) return null;

  return (
    <CenteredContainer>
      <CustomColumnContainer>
        <div>
          <h3>Current location</h3>
        </div>
        {currentLocation.city && currentLocation.country ? (
          <div>{`${currentLocation.city}, ${currentLocation.country}`}</div>
        ) : (
          <div>Browser based</div>
        )}
        <div>Lat: {currentLocation.lat} </div>
        <div>Long: {currentLocation.long} </div>
        <div>
          <button>Change location</button>
        </div>
      </CustomColumnContainer>
    </CenteredContainer>
  );
};

export default Location;
