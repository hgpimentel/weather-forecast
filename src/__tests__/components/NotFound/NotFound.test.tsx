import renderer from "react-test-renderer";
import { NotFound } from "../../../components";

describe("Not found component", () => {
  it("renders correctly", () => {
    const component = renderer
      .create(<NotFound message="Not Found!" />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
