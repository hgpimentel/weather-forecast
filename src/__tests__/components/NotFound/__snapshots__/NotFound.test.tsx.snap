// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not found component renders correctly 1`] = `
<div
  className="sc-gtsrHT jMpFrY"
>
  <h1>
    Not Found!
  </h1>
</div>
`;
